common/objectcachev2.h

```diff
@@ -58,15 +52,14 @@ class ObjectCacheV2 {
             : key(std::forward<KeyType>(key)), ref(nullptr) {}
 
         std::shared_ptr<V> update(std::shared_ptr<V> r, uint64_t ts = 0) {
-            SCOPED_LOCK(reflock);
             lastcreate = ts;
-            std::swap(r, ref);
+            r = std::atomic_exchange(&ref, r);
```

Comment:
why not simply atomic_store?

Reply:
atomic_store only stores but cannot load origin shared_ptr value for returning.

Reply:
Old value have to be load and return.

See line 128.

Reply:
The point is old object which ref refereed to should not be destruct during calling update or reset.

the shared_ptr have to be returned, so caller can decide if destruct right now or hold by other struct to destruct later.